app/uk/gov/hmrc/nisp/views/includes/language_selection.scala.html

```diff
@@ -0,0 +1,20 @@
+@import play.api.Play.current
+@import uk.gov.hmrc.urls.Link
+
+@(langMap  : Map[String, Lang], langToCall : (String => Call), customClass: Option[String] = None, appName: Option[String] = None)(implicit lang: Lang)
+
+<p class="@if(customClass.isDefined) {@customClass.get}">
+    @langMap.map { case (key: String, value: Lang) =>
+            @if(lang.code != value.code) {
+                @Link.toInternalPage(
+                    id      = Some(s"$key-switch"),
+                    url     = s"${langToCall(key)}",
+                    value   = Some(key.capitalize),
+                    dataAttributes = Some(Map("journey-click" -> s"${appName.fold("")(a=>a)}:language: ${value.code}"))
```

Comment:
Instead of```${appName.fold("")(a=>a)}``` use ```${appName.getOrElse("")}``` for strings, however:
I don't think you want to write out the data attribute if the app name is not passed in.
So you probably want something like:
```dataAttributes = appName.map(name => Map("journey-click" -> s"$name:language: ${value.code}")```